drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/test/java/org/drools/workbench/screens/scenariosimulation/client/handlers/AbstractScenarioSimulationDocksHandlerTest.java

```diff
@@ -0,0 +1,204 @@
+        final Collection<UberfireDock> docks = abstractScenarioSimulationDocksHandlerSpy.provideDocks("id");
+        final UberfireDock toolsDock = (UberfireDock) docks.toArray()[MANAGED_DOCKS.TOOLS.ordinal()];
+
+        abstractScenarioSimulationDocksHandlerSpy.expandToolsDock();
+
+        verify(authoringWorkbenchDocksMock).expandAuthoringDock(eq(toolsDock));
+    }
+
+    @Test
+    public void setScesimPath() {
+        final Collection<UberfireDock> docks = abstractScenarioSimulationDocksHandlerSpy.provideDocks("id");
+        final UberfireDock settingsDock = (UberfireDock) docks.toArray()[MANAGED_DOCKS.SETTINGS.ordinal()];
+        final UberfireDock toolsDock = (UberfireDock) docks.toArray()[MANAGED_DOCKS.TOOLS.ordinal()];
+        final UberfireDock cheatSheetDock = (UberfireDock) docks.toArray()[MANAGED_DOCKS.CHEATSHEET.ordinal()];
+        String TEST_PATH = "TEST_PATH";
+        abstractScenarioSimulationDocksHandlerSpy.setScesimEditorId(TEST_PATH);
+        assertTrue(settingsDock.getPlaceRequest().getParameters().containsKey(SCESIMEDITOR_ID));
+        assertEquals(TEST_PATH, settingsDock.getPlaceRequest().getParameter(SCESIMEDITOR_ID, "null"));
+        assertTrue(toolsDock.getPlaceRequest().getParameters().containsKey(SCESIMEDITOR_ID));
+        assertEquals(TEST_PATH, toolsDock.getPlaceRequest().getParameter(SCESIMEDITOR_ID, "null"));
+        assertTrue(cheatSheetDock.getPlaceRequest().getParameters().containsKey(SCESIMEDITOR_ID));
+        assertEquals(TEST_PATH, cheatSheetDock.getPlaceRequest().getParameter(SCESIMEDITOR_ID, "null"));
+    }
+
+    @Test
+    public void resetDocks() {
+        abstractScenarioSimulationDocksHandlerSpy.resetDocks();
+        verify(testToolsPresenterMock, times(1)).reset();
+        verify(settingsPresenterMock, times(1)).reset();
+        verify(cheatSheetPresenterMock, times(1)).reset();
+    }
+
+    @Test
+    public void addDocks() {
+        abstractScenarioSimulationDocksHandlerSpy.addDocks();
+        verify(abstractScenarioSimulationDocksHandlerSpy, times(1)).refreshDocks(eq(true), eq(false));
+    }
+
+    @Test
+    public void removeDocks() {
+        abstractScenarioSimulationDocksHandlerSpy.removeDocks();
+        verify(abstractScenarioSimulationDocksHandlerSpy, times(1)).refreshDocks(eq(true), eq(true));
+    }
+
+    @Test
+    public void getCurrentRightDockPlaceRequest() {
+        PlaceRequest placeRequest = abstractScenarioSimulationDocksHandlerSpy.getCurrentRightDockPlaceRequest("identifier");
+        assertNotNull(placeRequest);
+        assertEquals("identifier", placeRequest.getIdentifier());
+        assertNotNull(placeRequest.getParameter(SCESIMEDITOR_ID, ""));
+    }
+
+    @Test
+    public void getTestToolsView() {
+        TestToolsView testToolsViewMock = mock(TestToolsView.class);
+        AbstractWorkbenchActivity activityMock = mock(AbstractWorkbenchActivity.class);
+        when(activityMock.getWidget()).thenReturn(testToolsViewMock);
```

Comment:
How about adding a test to check 
getTestToolsView 
getCheatSheetView
getSettingsView
are empty when         activity is null() 
 when(activityMock.getActivity()).thenReturn(null);

Reply:
@dupliaka Done